sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/JsonSupport.java

```diff
@@ -130,11 +132,27 @@ public static <T> Any encodeJson(T value, String jsonType) {
    * the JSON string as bytes as value and a type URL starting with "json.kalix.io/".
    *
    * @param valueClass The type of class to deserialize the object to, the class must have the
-   *     proper Jackson annotations for deserialization.
+   *                   proper Jackson annotations for deserialization.
+   * @param any        The protobuf Any object to deserialize.
    * @return The decoded object
    * @throws IllegalArgumentException if the given value cannot be decoded to a T
    */
   public static <T> T decodeJson(Class<T> valueClass, Any any) {
+    return decodeJson(valueClass, any, Optional.empty());
+  }
+
+  /**
+   * Decode the given protobuf Any object to an instance of T using Jackson. The object must have
+   * the JSON string as bytes as value and a type URL starting with "json.kalix.io/".
+   *
+   * @param valueClass       The type of class to deserialize the object to, the class must have the
+   *                         proper Jackson annotations for deserialization.
+   * @param any              The protobuf Any object to deserialize.
+   * @param jacksonMigration The optional @{@link JsonMigration} implementation used for deserialization.
+   * @return The decoded object
+   * @throws IllegalArgumentException if the given value cannot be decoded to a T
+   */
+  public static <T> T decodeJson(Class<T> valueClass, Any any, Optional<? extends JsonMigration> jacksonMigration) {
```

Comment:
It is a problematic situation. Initially, this method was private, but to make it private I would need to move the `@Migration` to java-sdk-protobuf.

Reply:
I think it's better to have the Migration annotation in the java-sdk-protobuf package, no? 

Or do we plan to use this method in other contexts? 

The `decodeJson` method is also used when consuming json from a topic, but in such case, I doubt that it will be useful to pass a migration class as the type URL won't have the version number attached to it.